src/RestApi/StoreApi/Schemas/CheckoutSchema.php

```diff
@@ -40,16 +63,23 @@ public function get_properties() {
 	/**
 	 * Return the response for checkout.
 	 *
-	 * @todo we need to determine the fields we want to return after processing e.g. redirect URLs.
+	 * @throws Exception On invalid response.
 	 *
-	 * @param object $checkout_result Result from checkout action.
+	 * @param array $item Results from checkout action.
 	 * @return array
 	 */
-	public function get_item_response( $checkout_result ) {
-		$order_schema = new OrderSchema();
+	public function get_item_response( $item ) {
+		$order          = wc_get_order( absint( $item['order_id'] ) );
+		$payment_result = $item['payment_result'] instanceof PaymentResult ? $item['payment_result'] : false;
+
+		if ( ! $order || ! $payment_result ) {
+			throw new Exception( 'Invalid response.' );
+		}
+
 		return [
-			'note'  => 'This is a placeholder',
-			'order' => $order_schema->get_item_response( $checkout_result['order'] ),
+			'payment_status'  => $payment_result->status,
+			'payment_details' => $payment_result->payment_details,
+			'order'           => ( new OrderSchema() )->get_item_response( $order ),
```

Comment:
If payment method returns a redirect url, should we require a standard key for that and return that on the root of the response object so that checkout processing can use that to know where to redirect to?

Reply:
Possibly but I'm not sure if we want to enforce it for all gateways as as standard key. We may for instance what to simply default to the thanks page, or replace that at some point. We should revisit this after implementing some gateways.